src/client/src/ui/analytics/components/AnalyticsLineChart.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import moment from 'moment'
+import numeral from 'numeral'
+import { XAxis, YAxis, ResponsiveContainer, LineChart, Line, ReferenceLine, Tooltip } from 'recharts'
+import { ToolTipStyle, ToolTipChildRight, ToolTipChildLeft } from './styled'
+
+interface LineChartProps {
+  seriesData: PricePoint[]
+}
+
+interface PricePoint {
+  x: Date
+  y: string
+}
+
+interface DataPoint {
+  x: string
+  y: string
+}
+
+const tickFormatYAxis = (x: string) => numeral(x).format('0.0a')
+
+const getGradientOffset = (data: DataPoint[]) => {
```

Comment:
likewise

Reply:
Also I'm not a fan of this function (sorry!) or the name.  It returns a value that is used as an offset but what this function actually does is return a ratio of Max(val) / min-max difference.  I feel it bloats the component a bit and could perhaps live elsewhere.